container/docker/handler.go

```diff
@@ -294,32 +294,8 @@ func (self *dockerContainerHandler) GetStats() (*info.ContainerStats, error) {
 }
 
 func (self *dockerContainerHandler) ListContainers(listType container.ListType) ([]info.ContainerReference, error) {
-	if self.name != "/docker" {
```

Comment:
In all the tests I've done, it doesn't. I have tested this on Ubuntu Precise (upstart), Debian Jessie/Sid (sysv init) and Debian Jessie (systemd).

When the manager is started it registers the `/` container and starts searching for containers using `detectSubcontainers("/")`. Since `/` is technically a raw container, all Docker containers are discovered using their entries in the cgroup mount. 

I have not found any place where the `dockerContainerHandler` with the name `/docker` is created. Because it doesn't exist, this **if** will never be **false**